libs/colbert/ragstack_colbert/base_embedding_model.py

```diff
@@ -62,3 +62,21 @@ def embed_query(
         Returns:
             Tensor: A tensor representing the embedded query.
         """
+
+    @abstractmethod
+    def optimized_query_embeddings(
```

Comment:
seems odd to put this into the interface... we should figure out which method works better and just make it the default for `embed_query` above.